socs/agents/ld_monitor/agent.py

```diff
@@ -0,0 +1,426 @@
+                    'clear': data_split[4],
+                    'r_timer': data_split[5],
+                    'o_timer': data_split[6],
+                    'y_timer': data_split[7],
+                    'g_timer': data_split[8],
+                    'allclear_timer': data_split[9],
+                    'faultcode': faultcode
+                }
+
+                self.data_dict.update(self.newdata_dict)
+                return self.data_dict
+
+            # disregard "alarm timers" sentence, but update sentence type
+            elif param == 'WT':
+                self.newdata_dict = {'d_type': 4}
+                self.data_dict.update(self.newdata_dict)
+                return self.data_dict
+
+    def read_cycle(self):
+        """
+        In each cycle data is read and then parsed following
+        the format required to publish data to the ocs feed
+        """
+        try:
+            self.read_data()
+
+            # updates time since last strike if strike data exists
+            if self.data_dict['time_last'] == -1.:
+                self.data_dict['tsince_last'] = -1.
+            else:
+                self.data_dict['tsince_last'] = (time.time()
+                                                 - self.data_dict['time_last'])
+
+            return self.data_dict
+
+        except BaseException:
+            pass
+            self.log.info('LD data read error, passing to next data iteration')
+
+
+class LDMonitorAgent:
+    """Monitor the Lightning Detector data via UDP.
+
+    Parameters
+    ----------
+    agent : OCSAgent
+        OCSAgent object which forms this Agent
+    sample_interval : float
+        Time between samples in seconds.
+
+    Attributes
+    ----------
+    agent : OCSAgent
+        OCSAgent object which forms this Agent
+    take_data : bool
+        Tracks whether or not the agent is actively issuing SNMP GET commands
+        to the ibootbar. Setting to false stops sending commands.
+    log : txaio.tx.Logger
+        txaio logger object, created by the OCSAgent
+    """
+
+    def __init__(self, agent, sample_interval=15.):
+
+        self.agent: ocs_agent.OCSAgent = agent
+        self.log = agent.log
+        self.lock = TimeoutLock()
+
+        self.pacemaker_freq = 1. / sample_interval
+
+        self.initialized = False
+        self.take_data = False
+
+        self.LDMonitor = None
+
+        agg_params = {
+            'frame_length': 10 * 60  # [sec]
+        }
+        self.agent.register_feed('LDMonitor',
+                                 record=True,
+                                 agg_params=agg_params,
+                                 buffer_time=0)
+
+    def _connect(self):
+        """connect()
+        Instantiates LD object and check if client is open
+        """
+        self.LDMonitor = LDMonitor()
+        self.initialized = True
+
+    @ocs_agent.param('auto_acquire', default=False, type=bool)
+    def init_LDMonitor(self, session, params=None):
```

Comment:
This should stay as `init_ld_monitor` as it was before the updates.

The docstring should also start with:
```
"""init_ld_monitor(auto_acquire=False)

**Task** - ...
```

Reply:
Sorry, I had to prioritize other tasks on my last shift, but I'll work on the changes and docs page on the next week. I'll let you know if something comes up. Thanks!

Reply:
Thanks for adding a docs page! I moved it into the right place and added it to the index so it'll show up in the sidebar. While I was doing that I saw a few warnings from the build, so I addressed those and cleaned up some of the formatting. You can see the page in builds from this PR here: https://socs--640.org.readthedocs.build/en/640/agents/ld_monitor.html

To find that later, click "Details" next to the "docs/readthedocs.org:socs" check down below.